### ISSUE TITLE
Add Content-MD5 header when staging blocks

### ISSUE THREAD

**Issue by someone1** [Azure/azure-storage-blob-go#56]
It's trivial to calculate the md5 sum for the chunk provided (do it [here](https://github.com/Azure/azure-storage-blob-go/blob/master/2018-03-28/azblob/zz_generated_block_blob.go#L270)?):

```go
        chunk, err := ioutil.ReadAll(body)
        if err != nil {
                return req, pipeline.NewError(err, "failed to read request body")
        }
        body.Seek(0, io.SeekStart)
        md5sum := md5.Sum(chunk)
	req.Header.Set("Content-MD5", base64.StdEncoding.EncodeToString(md5sum[:]))
```

Or allow the user to pass the option to include the header when staging the block

**Comment by JeffreyRichter** [comment]
Yes, StageBlock and Upload could allow you to pass the Content-MD5 value. We will consider adding this in a future release. The SDK will not calculate it for you automatically (for several reasons; one of which is the performance hit). 

I am curios though: Why do you think using the MD5 is useful? TCP alrady includes checksum values to ensure that bits travel safely over the network and calculating MD5 is a huge performance hit. In addition, it is possible that a bit has flipped BEFORE you calculate the MD5 hash and now you are guaranteed to persist bad data. 

So, this dubious value of MD5 is why our SDK currently does not support MD5 checking. But, I'd love to hear your feedback about this in case I'm missing something.

**Comment by someone1** [comment]
I realize the potential redundancy here. 

I will be using it in a situation where the value is calculated in memory (server is using ECC RAM), but the file is dumped to disk before being uploaded.

I assume the option was provided by the Azure API creators with some use-case in mind and data integrity is key for my use case, so I'd like to assume the worst and leverage whatever I can to ensure my bits arrive safely and as intended.

**Comment by zezha-msft** [comment]
This was fixed in [version 0.2.0](https://github.com/Azure/azure-storage-blob-go/releases/tag/0.2.0).
